scylla-persistence/src/main/java/com/netflix/conductor/scylla/dao/ScyllaExecutionDAO.java

```diff
@@ -331,38 +342,67 @@ public List<TaskModel> createTasks(List<TaskModel> tasks) {
      * @method to add the task_in_progress table with the status of the task if task is not already present
      */
     public void addTaskInProgress(TaskModel task) {
-        ResultSet resultSet =
-                session.execute(
-                        selectTaskInProgressStatement.bind(task.getTaskDefName(),
-                                UUID.fromString(task.getTaskId())));
-        if (resultSet.all().isEmpty() || resultSet.all().size()<1) {
-            session.execute(
-                    insertTaskInProgressStatement.bind(task.getTaskDefName(),
-                            UUID.fromString(task.getTaskId()),
-                            UUID.fromString(task.getWorkflowInstanceId()),
-                            true));
+        ResultSet resultSet = getTaskInProgressTaskFromTaskModel(task);
+        if (CollectionUtils.isEmpty(resultSet.all())) {
+            session.execute(insertTaskInProgressV2Statement.bind(task.getTaskDefName(), UUID.fromString(task.getTaskId()),
+                    UUID.fromString(task.getWorkflowInstanceId()), true));
+        } else {
+            LOGGER.info("Task with defName {} and Id {} and status {} in addTaskInProgress NOT inserted as already exists  ", task.getTaskDefName(),
```

Comment:
A slightly better way to log the task is to override toString in Task Entity and just use this in the logger

Reply:
this is existing conductor code, will try to refactor during migration to conductor-oss